### PR TITLE
Don't allow address and signature to be duplicated.

### PR DESCRIPTION
**Pull request by raress96**

### CODE REVIEW

File: interact/index.ts [hunk 1]
```diff
@@ -270,9 +270,9 @@ program.command("updateSdkCore").action(async () => {
   const decodedAttributes = decoded.valueOf();
 
   const contractPriceData = {
-    hearbeat: decodedAttributes.hearbeat.toNumber(),
-    timestamp: decodedAttributes.timestamp.toNumber(),
-    price: decodedAttributes.price.toNumber(),
+    hearbeat: decodedAttributes?.hearbeat?.toNumber(),
```
Comment (DZariusz) [inline interact/index.ts:273]: just to be on the same page: there should be no case, where this values does not exists, if record is empty they all should be 0

File: tests/contract.test.ts [hunk 2]
```diff
@@ -262,20 +266,115 @@ test("Update invalid signer", async () => {
...
+
+      e.U32(1),
+      e.Tuple(
+        e.U32(priceData.hearbeat),
+        e.U32(priceData.timestamp),
+        e.U(priceData.price.toNumber()),
+      ),
+
+      e.U32(15),
+      e.Tuple(
```
Comment (DZariusz) [inline tests/contract.test.ts:318]: imo we can simplify this test and did only two checks:
1. with [signatureBob, signatureBob] - it should fail
2. [signatureBob, signatureBobCarol] and [signatureBobCarol, signatureBob] - one should pass and should fail

but this is not a blocker.

File: umbrella-feeds/src/lib.rs [hunk 3]
```diff
@@ -182,10 +182,12 @@ pub trait UmbrellaFeeds: proxy::ProxyModule {
 
             self.verify_signature(&hash, &raw_signature);
 
+            require!(validators.find(&raw_signature.address).is_none(), "Signatures out of order");
```
Comment (DZariusz) [inline umbrella-feeds/src/lib.rs:185]: let's discuss this on discord, if we can do better.

**Review by DZariusz — APPROVED** [review]
comments I left are not blockers for marge.

File: interact/index.ts (hunk 1 above)
Comment (raress96) [reply interact/index.ts:273]: Yes, this is just for testing.

File: tests/contract.test.ts (hunk 2 above)
Comment (raress96) [reply tests/contract.test.ts:318]: It could be simplified, but I left it like this to also check that the max gas it uses is 53M in this case.

**Pull request merged**